training_controller/main.py

```diff
@@ -4,134 +4,205 @@
-    """
-    script = f"ctx._source.status = '{job_status}';"
-    docs_to_update = [
-        {
-            "_id": request_id,
-            "_op_type": op_type,
-            "_index": index,
-            "script": script,
-        }
-    ]
-    logging.info(f"ES job {request_id} status update : {job_status}")
+def get_gpu_status():
     try:
-        async for ok, result in async_streaming_bulk(es, docs_to_update):
-            action, result = result.popitem()
-            if not ok:
-                logging.error(f"failed to index documents {docs_to_update}")
+        results = requests.get("http://opni-internal:11080/ModelTraining/gpu_info")
+        decoded_result = json.loads(results.content.decode())
+        if "list" in decoded_result:
+            gpu_resources_list = decoded_result["list"]
+            for gpu_info in gpu_resources_list:
+                if int(gpu_info["allocatable"]) > 0:
+                    return "available"
+        return "unavailable"
+
     except Exception as e:
         logging.error(e)
+        return "unavailable"
 
 
-async def es_training_signal():
-    """
-    collect job training signal from elasticsearch, and add to job queue.
-    """
-    query_body = {"query": {"bool": {"must": {"match": {"status": "submitted"}}}}}
-    index = "training_signal"
-    current_time = int(datetime.timestamp(datetime.now()))
-    job_payload = {
-        "model_to_train": "nulog",
-        "time_intervals": [
-            {
-                "start_ts": (current_time - DEFAULT_TRAINING_INTERVAL) * (10 ** 9),
-                "end_ts": current_time * (10 ** 9),
-            }
-        ],
-    }
-    signal_index_exists = False
+async def get_gpu_service_status():
     try:
-        signal_index_exists = await es.indices.exists(index)
-        if not signal_index_exists:
-            signal_created = await es.indices.create(index=index)
-    except exceptions.TransportError as e:
-        logging.error(e)
-    while True:
-        try:
-            user_signals_response = await es.search(
-                index=index, body=query_body, size=100
-            )
-            user_signal_hits = user_signals_response["hits"]["hits"]
-            if len(user_signal_hits) > 0:
-                for hit in user_signal_hits:
-                    training_job_payload = {
-                        "source": "elasticsearch",
-                        "_id": hit["_id"],
-                        "model": "nulog-train",
-                        "payload": job_payload,
-                    }
-                    await update_es_job_status(
-                        request_id=hit["_id"], job_status="training scheduled"
-                    )
-                    await schedule_training_job(training_job_payload)
-        except (exceptions.NotFoundError, exceptions.TransportError) as e:
-            logging.error(e)
-        await asyncio.sleep(60)
+        response = await nw.request("gpu_service_running", b"check-up", timeout=1)
```

Comment:
what happens If the GPU service is in the middle of a training job?

Reply:
it returns unavailable then. This case is more for if the GPU is removed and the pod is pending but it'll also return unavailable in addition to the gpu_training_request variable being set to 1.

Reply:
cool. as long as its an expected behavior then all good.